bundles/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/PlugwiseBinding.java

```diff
@@ -68,142 +74,170 @@ protected void removeBindingProvider(PlugwiseBindingProvider bindingProvider) {
         super.removeBindingProvider(bindingProvider);
     }
 
-    @SuppressWarnings("rawtypes")
     @Override
-    public void updated(Dictionary config) throws ConfigurationException {
+    public void updated(Dictionary<String, ?> config) throws ConfigurationException {
 
-        if (config != null) {
+        if (config == null) {
+            return;
+        }
 
-            // First of all make sure the Stick gets set up
-            Enumeration keys = config.keys();
-            while (keys.hasMoreElements()) {
+        validateKeyPatternsInConfig(config);
 
-                String key = (String) keys.nextElement();
+        stick = setupStick(config);
 
-                // the config-key enumeration contains additional keys that we
-                // don't want to process here ...
-                if ("service.pid".equals(key)) {
-                    continue;
-                }
+        if (stick != null) {
+            setupNonStickDevices(config);
+            setProperlyConfigured(true);
+        } else {
+            logger.error("Plugwise needs at least one Stick in order to operate");
```

Comment:
This would be better as a `logger.warn`, since `logger.error` is reserved for messages that affect system stability.